packages/core/src/request.ts

```diff
@@ -95,7 +95,13 @@ export class Request {
     headers?: Record<string, string>;
 
     /** Custom user data assigned to the request. */
-    userData: Record<string, any>;
+    userData: Record<string, any> = {};
+
+    /** Private store for the custom user data assigned to the request. */
+    #userData: Record<string, any>;
```

Comment:
i would rather not use native private properties, they are very slow in node < 18 (and working with request class is indeed a hot path for us), as well as have some pitfalls (e.g. you can't create instance of such class via `Object.create()`)

also, what is the monivation for those extra properties? i would rather keep the previous solution if this is just about "hiding the __crawlee key", as that can be easily done by making it non enumerable, and we will have much simpler and more performant solution

Reply:
This was a lot of trial and error (more error than trial), but non-enumerable properties seem like a solution to many of the problems I've encountered. Big thanks for the tips! 💡